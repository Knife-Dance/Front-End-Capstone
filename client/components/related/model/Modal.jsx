import React, {useState, useEffect, useContext} from 'react';
import {dataUrl, related} from '../sample-data';
import css from './modal.module.css'



const Modal = ({setShowModal, productData, cardData}) => {

  let dataA, dataB, featureArr;

  if (productData && cardData) {

    dataA = productData.features.map((each) => {
      return `${each.value || ''} ${each.feature || ''}`
    });

    dataB= cardData.features.map((each) => {
      return `${each.value || ''} ${each.feature || ''}`
    });

     featureArr = [...dataA, ...dataB];

    featureArr = new Set(featureArr);

  }




  return (
    <div >
      { productData ? (
        <div className={css.modalBox}>
          <div className={css.modalContainer}>
          <button onClick={() => setShowModal(prev => !prev)}className={css.closeModalBtn}>X</button>

          <div >
            <h3>Comparing</h3>
            <div className={css.tableHeader}>
              <p >{productData.name}</p>
              <p>{cardData.name}</p>
            </div>

            <table>

              {Array.from(featureArr).map((item, key) => (
                <>
                <tr key={key}>
                <td className={css.leftCheck}>{dataA.includes(item) ? <i className={"fas fa-check"}></i> : ''}</td>

                  <td>{item}</td>
                  <td className={css.rightCheck}>{dataB.includes(item) ? <i className={"fas fa-check"}></i> : ''}</td>
                </tr>
                </>
              ))}

            </table>

          </div>
          </div>
          </div>
          ): null}

    </div>
  )

}



export default Modal;


