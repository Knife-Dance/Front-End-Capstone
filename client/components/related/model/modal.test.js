// import 'jsdom-global/register';
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import sinon from 'sinon'
import Modal from './Modal';
import Relate from '../product-card/Related-product';

Enzyme.configure({ adapter: new Adapter() });

let productDataExample = {
  "id": 17067,
  "campus": "hr-rfp",
  "name": "Camo Onesie",
  "slogan": "Blend in to your crowd",
  "description": "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  "category": "Jackets",
  "default_price": "140.00",
  "created_at": "2021-02-23T04:22:44.728Z",
  "updated_at": "2021-02-23T04:22:44.728Z",
  "features": [
      {
          "feature": "Fabric",
          "value": "Canvas"
      },
      {
          "feature": "Buttons",
          "value": "Brass"
      }
  ]
}


const cardDataExample = {
  "id": 17068,
  "campus": "hr-rfp",
  "name": "Bright Future Sunglasses",
  "slogan": "You've got to wear shades",
  "description": "Where you're going you might not need roads, but you definitely need some shades. Give those baby blues a rest and let the future shine bright on these timeless lenses.",
  "category": "Accessories",
  "default_price": "69.00",
  "created_at": "2021-02-23T04:22:44.728Z",
  "updated_at": "2021-02-23T04:22:44.728Z",
  "features": [
      {
          "feature": "Lenses",
          "value": "Ultrasheen"
      },
      {
          "feature": "UV Protection",
          "value": null
      },
      {
          "feature": "Frames",
          "value": "LightCompose"
      }
  ]
}

describe('Modal', () => {

  it('Renders Modal component', () => {
    const wrapper = shallow( <Modal/>);;
    expect(wrapper).toMatchSnapshot()
  });

  it('Modal should contain a table', () => {
    const wrapper = shallow( <Modal/>);
    expect(wrapper.find('table')).toBeTruthy();
  });

  it('Render button', () => {
    const wrapper = mount( <Modal setShowModal={true} productData={productDataExample} cardData ={cardDataExample}/>);
    const btn = wrapper.find('h3').text()
    expect(btn).toEqual('Comparing')

    //Ask the text()
    // const btn = wrapper.find('.closeModalBtn').text()
    // expect(btn).toBeTruthy().toEqual('X')
  });

});


