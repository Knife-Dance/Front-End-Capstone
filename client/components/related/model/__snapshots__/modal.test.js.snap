// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Renders Modal component 1`] = `ShallowWrapper {}`;
